app/lib/actions/candlepin/import_pool_handler.rb

```diff
@@ -62,10 +64,34 @@ def remove_pool(pool_id)
         end
       end
 
+      def get_consumer_uuid(message_content)
+        parsed = JSON.parse(message_content)
+        parsed['consumer']['uuid']
+      end
+
+      def index_pool_on_host(message, create)
+        content = message.content
+        host = create ? content['newEntity'] : content['oldEntity']
```

Comment:
could you just try to get it from either newEntity or oldEntity, whichever is there and provides it?

something like:

uuid = get_consumer_uuid(content['newEntity']) || get_consumer_uuid(content['oldEntity']) 

you'd have to change get_consumer_uuid to handle nil's a bit better, but that create argument to the method just feels dirty :)